app/controllers/settings_controller.rb

```diff
@@ -256,6 +256,25 @@ def twitter_disconnect
     return redirect_to "/settings"
   end
 
+  def keybase_auth
+    if !params[:kb_username].present? || !params[:kb_signature].present? || !params[:kb_ua].present? || (!params[:username].present? || !params[:username] == @user.username)
+      flash[:error] = "Invalid Keybase connection parameters"
+      return redirect_to "/settings"
+    end
+
+    kb_username = params[:kb_username]
+    kb_signature = params[:kb_signature]
+    kb_ua = params[:kb_ua]
+
+    if Keybase.validate_initial(kb_username, kb_signature, @user.username)
+      @user.keybase_signatures = [{ kb_username: kb_username, sig_hash: kb_signature }]
```

Comment:
Please filter these response values down to alphanumeric if possible - we print them later and I'd like to make sure there's no XSS injection potential. I'd be less paranoid on the way in if Rails would let us tag attributes as user-supplied and really freak out if they ever get to output, but that's not a feature sooo....

Reply:
I've never used ruby before, how would I go about doing this?

Reply:
I hadn't read the `Keybase` class before I wrote this, and it would be a good place to enforce the validity of this data. `kb_username.gsub!(/[^0-9A-Za-z_\-]/, '')` removed anything that's not an acceptable character in the `kb_username` variable.